### PR TITLE
Small cleanup in TeradataDateFunctions

### PR DESCRIPTION
**Pull request by findepi**
extracted from https://github.com/prestodb/presto/pull/10128

### CODE REVIEW

File: presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/TeradataDateFunctions.java [hunk 1]
```diff
@@ -78,19 +77,11 @@ public static Slice toChar(
...
-                    MethodHandle castToDate,
-            ConnectorSession session,
-            @SqlType(StandardTypes.VARCHAR) Slice dateTime,
-            @SqlType(StandardTypes.VARCHAR) Slice formatString)
+    public static long toDate(ConnectorSession session, @SqlType(StandardTypes.VARCHAR) Slice dateTime, @SqlType(StandardTypes.VARCHAR) Slice formatString)
     {
         try {
-            long millis = parseMillis(session, dateTime, formatString);
-            return (long) castToDate.invokeExact(session, millis);
+            long millis = parseMillis(UTC_KEY, session.getLocale(), dateTime, formatString);
```
Comment (sopel39) [inline presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/TeradataDateFunctions.java]: Shouldn't passing `UTC_KEY` depend on legacy timestamp flag?

Comment (sopel39) [reply presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/TeradataDateFunctions.java]: No new tests are needed?

File: presto-teradata-functions/src/test/java/com/facebook/presto/teradata/functions/TestTeradataDateFunctions.java [hunk 2]
```diff
@@ -152,7 +147,7 @@ private void assertDate(String projection, int year, int month, int day)
         assertFunction(
                 projection,
                 DateType.DATE,
-                sqlDate(new DateTime(year, month, day, 0, 0, DateTimeZone.UTC)));
+                new SqlDate(toIntExact(LocalDate.of(year, month, day).toEpochDay())));
```
Comment (sopel39) [inline presto-teradata-functions/src/test/java/com/facebook/presto/teradata/functions/TestTeradataDateFunctions.java]: Shouldn't commit message be:
`Fix TestTeradataDateFunctions to work with non-legacy timestamps and in zones with negative offset`?

File: presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/TeradataDateFunctions.java (hunk 1 above)
Comment (findepi) [reply presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/TeradataDateFunctions.java]: > Shouldn't passing UTC_KEY/session zone depend on legacy timestamp flag?

DATEs have just one semantics (no legacy here)

> No new tests are needed?

Tests exists and implementation was working correctly. This just simplifies.

File: presto-teradata-functions/src/test/java/com/facebook/presto/teradata/functions/TestTeradataDateFunctions.java (hunk 2 above)
Comment (findepi) [reply presto-teradata-functions/src/test/java/com/facebook/presto/teradata/functions/TestTeradataDateFunctions.java]: I don't think so.

**Pull request merged**